tests/foreman/longrun/test_oscap.py

```diff
@@ -300,6 +318,134 @@ def test_positive_push_updated_content(self):
+                tailoring_path=tailoring_path,
+            )
+            self.assertIsNotNone(
+                self.oscaptailoringfile.search(tailoringfile_name))
+            # Creates oscap_policy for rhel7.
+            make_oscappolicy(
+                session,
+                content=policy_values.get('content'),
+                host_group=policy_values.get('hgrp'),
+                name=policy_values.get('policy'),
+                period=OSCAP_PERIOD['weekly'],
+                profile=OSCAP_PROFILE['common'],
+                period_value=OSCAP_WEEKDAY['friday'],
+                tailoring=tailoringfile_name,
+                tailoring_profile=OSCAP_PROFILE['tailoring_rhel7']
+            )
+            distro_os = vm_values.get('distro')
+            with VirtualMachine(distro=distro_os) as vm:
+                vm.install_katello_ca()
+                vm.register_contenthost(
+                    self.config_env['org_name'],
+                    self.config_env['ak_name'].get(distro_os)
+                )
+                self.assertTrue(vm.subscribed)
+                vm.configure_puppet(rhel7_repo)
+                self.hosts.update(
+                    name=vm._target_image,
+                    domain_name=vm.domain,
+                    parameters_list=[
+                        ['Host', 'Host Group', vm_values.get('hgrp')],
+                        ['Host', 'Lifecycle Environment',
+                            self.config_env['env_name']],
+                        ['Host', 'Content View', self.config_env['cv_name']],
+                        ['Host', 'Reset Puppet Environment', True],
+                        ['Host', 'Openscap Capsule',
+                            self.config_env['sat6_hostname']],
+                    ],
+                )
+                session.nav.go_to_hosts()
+                self.hosts.update_host_bulkactions(
+                    ['{0}.{1}'.format(vm._target_image.lower(), vm.domain)],
+                    action='Assign Compliance Policy',
+                    parameters_list=[
+                        {'policy': policy_values.get('policy')}],
+                )
+                # Run "puppet agent -t" twice so that it detects it's,
+                # satellite6 and fetch katello SSL certs.
+                for _ in range(2):
+                    vm.run(u'puppet agent -t 2> /dev/null')
+                result = vm.run(
+                    u'cat /etc/foreman_scap_client/config.yaml'
+                    '| grep profile'
+                )
+                self.assertEqual(result.return_code, 0)
+                # Runs the actual oscap scan on the vm/clients and
+                # uploads report to Internal Capsule.
+                vm.execute_foreman_scap_client()
+                # Assert whether oscap reports are uploaded to
+                # Satellite6.
+                self.assertTrue(self.oscapreports.search(
+                    '{0}.{1}'.format(vm._target_image.lower(), vm.domain)))
```

Comment:
same question as above why do we need to reconstruct the vm host, why not simply vm.hostname